core/src/main/java/org/apache/iceberg/view/BaseView.java

```diff
@@ -103,4 +105,29 @@ public UpdateLocation updateLocation() {
   public UUID uuid() {
     return UUID.fromString(ops.current().uuid());
   }
+
+  /**
+   * This implementation of sqlFor will resolve what is considered the "closest" dialect. If an
+   * exact match is found, then that is returned. Otherwise, the first representation would be
+   * returned. If no SQL representation is found, null is returned.
+   */
+  @Override
+  public SQLViewRepresentation sqlFor(String dialect) {
+    Preconditions.checkArgument(dialect != null, "Invalid dialect: null");
+    Preconditions.checkArgument(!dialect.isEmpty(), "Invalid dialect: empty");
+    List<SQLViewRepresentation> sqlViewRepresentations =
+        currentVersion().representations().stream()
+            .filter(
+                viewRepresentation -> viewRepresentation.type().equals(ViewRepresentation.Type.SQL))
+            .map(viewRepresentation -> (SQLViewRepresentation) viewRepresentation)
+            .collect(Collectors.toList());
+    if (sqlViewRepresentations.isEmpty()) {
+      return null;
+    }
+
+    return sqlViewRepresentations.stream()
+        .filter(sqlViewRepresentation -> sqlViewRepresentation.dialect().equalsIgnoreCase(dialect))
+        .findFirst()
+        .orElseGet(() -> sqlViewRepresentations.get(0));
```

Comment:
Minor: To me, it seems like this is quite a bit of complexity to so something that's simple with a for loop:

```java
  SQLViewRepresentation closest = null;
  for (ViewRepresentation rep : currentVersion().representations()) {
    if (rep instanceof SQLViewRepresentation) {
      SQLViewRepresentation sql = (SQLViewRepresentation) rep;
      if (dialect.equalsIgnoreCase(sql.dialect())) {
        return sql;
      } else if (null == closest) {
        closest = sql
      }
    }
  }

  return closest;
```

Reply:
I agree, I took a second look at this code and realized this is a case where streams probably isn't that helpful